### ISSUE TITLE
set up github workflow

### ISSUE THREAD

**Issue by fishsticks89** [Unionjackjz1/EZ-Template#8]

**Comment by RishiKumarRay** [comment]
can you elaboarte please ?

**Comment by fishsticks89** [comment]
The idea was to automate the creation of releases with workflows, just as the testing is automated with the pros-build action. If @Unionjackjz1 wants me to do this, I can, but every release to this point seems to have gone smoothly.

**Comment by Unionjackjz1** [comment]
> The idea was to automate the creation of releases with workflows, just as the testing is automated with the pros-build action. If @Unionjackjz1 wants me to do this, I can, but every release to this point seems to have gone smoothly.

yes i think we should do this.  to the audience this library is attracting, using terminal seems like a scary thing.  

hotel added a tutorial to the [pros conductor page](https://pros.cs.purdue.edu/v5/cli/conductor.html) on how to make remote depos.  2.2.0 is currently being worked on, should we aim for this to be done for that release?  or hold it off until the release at the start of the new season.

**Comment by fishsticks89** [comment]
To be clear, this doesn't change anything for the user, it would only make the creation of releases like [this](https://github.com/EZ-Robotics/EZ-Template/releases/tag/v2.1.1) automatic when pushing to the main branch.

**Comment by fishsticks89** [comment]
It could also make deployment to a depot automatic

**Comment by ssejrog** [comment]
Partial status update: CI already automates the parts this issue's discussion converged on being lower-risk — every push builds the library, runs the host test suite, checks it warning-clean, and uploads the template as a build artifact (see .github/workflows/main.yml).

What's still manual, and what this issue originally asked for, is auto-cutting the actual GitHub Release (with notes and the packaged zip) on push to main. That's release packaging, and it's intentionally not being worked on right now — deferring rather than building it.
